Mathlib/Data/UInt.lean

```diff
@@ -97,7 +97,14 @@ run_cmd
       @[simp] lemma mk_val_eq : ∀ (a : $typeName), mk a.val = a
       | ⟨_, _⟩ => rfl
 
-      instance : CommRing $typeName :=
+      /--
+      This is not an instance; it is not needed in Mathlib,
+      and moreover results in a `Nat → $typeName` coercion which breaks assumptions made in the
```

Comment:
This isn't true; the coercion is defined higher up in the file; this just proves it's lawful.

So if we go down this route there are three instances to disable for each type, not one. I think asking the user to refer to these individually by name is unreasonable, and so we should scope them.